scripts/files/tests/fs_s3_test.py

```diff
@@ -124,3 +124,20 @@ def test_exists_object_starting_with_not_exists() -> None:
     file_exists = exists("s3://testbucket/hello/another.fi")
 
     assert file_exists is False
+
+
+@mock_s3  # type: ignore
+def test_list_files_in_uri() -> None:
+    s3 = resource("s3", region_name=DEFAULT_REGION_NAME)
+    boto3_client = client("s3", region_name=DEFAULT_REGION_NAME)
+    s3.create_bucket(Bucket="testbucket")
+    boto3_client.put_object(Bucket="testbucket", Key="data/collection.json", Body=b"test content")
```

Comment:
The bucket name should probably be pulled out into a variable, to make sure it's clear that it's a re-used value.

As for the `Body`, we don't actually care what the contents are, so we could either use an `any_body()`, `ANY_BODY`, or an empty `b""`.